Build failure: softether

### Nixpkgs version

Please select a version.

### Steps to reproduce

This command fails for me:
`nix build nixpkgs#softether`

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/286034818

### Relevant log output

```console
--snip warnings--
gcc -DNDEBUG -DVPN_SPEED -DUNIX -DUNIX_LINUX -DCPU_64 -D_REENTRANT -DREENTRANT -D_THREAD_SAFE -D_THREADSAFE -DTHREAD_SAFE -DTHREADSAFE -D_FILE_OFFSET_BITS=64 -I./src/ -I./src/Cedar/ -I./src/Mayaqua/ -O2 -fsigned-char -m64 -c src/Mayaqua/Secure.c -o tmp/objs/Mayaqua/Secure.o
[01m[Ksrc/Mayaqua/Secure.c:[m[K In function '[01m[KSignSecByObject[m[K':
[01m[Ksrc/Mayaqua/Secure.c:420:73:[m[K [01;31m[Kerror: [m[Kpassing argument 5 of '[01m[Ksec->Api->C_Sign[m[K' from incompatible pointer type [[01;31m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wincompatible-pointer-types-Wincompatible-pointer-types]8;;[m[K]
  420 |         ret = sec->Api->C_Sign(sec->SessionId, hash, sizeof(hash), dst, [01;31m[K&size[m[K);
      |                                                                         [01;31m[K^~~~~[m[K
      |                                                                         [01;31m[K|[m[K
      |                                                                         [01;31m[KUINT * {aka unsigned int *}[m[K
[01m[Ksrc/Mayaqua/Secure.c:420:73:[m[K [01;36m[Knote: [m[Kexpected '[01m[KCK_ULONG_PTR[m[K' {aka '[01m[Klong unsigned int *[m[K'} but argument is of type '[01m[KUINT *[m[K' {aka '[01m[Kunsigned int *[m[K'}
[01m[Ksrc/Mayaqua/Secure.c:424:81:[m[K [01;31m[Kerror: [m[Kpassing argument 5 of '[01m[Ksec->Api->C_Sign[m[K' from incompatible pointer type [[01;31m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wincompatible-pointer-types-Wincompatible-pointer-types]8;;[m[K]
  424 |                 ret = sec->Api->C_Sign(sec->SessionId, hash, sizeof(hash), dst, [01;31m[K&size[m[K);
      |                                                                                 [01;31m[K^~~~~[m[K
      |                                                                                 [01;31m[K|[m[K
      |                                                                                 [01;31m[KUINT * {aka unsigned int *}[m[K
[01m[Ksrc/Mayaqua/Secure.c:424:81:[m[K [01;36m[Knote: [m[Kexpected '[01m[KCK_ULONG_PTR[m[K' {aka '[01m[Klong unsigned int *[m[K'} but argument is of type '[01m[KUINT *[m[K' {aka '[01m[Kunsigned int *[m[K'}
[01m[Ksrc/Mayaqua/Secure.c:[m[K In function '[01m[KWriteSecKey[m[K':
[01m[Ksrc/Mayaqua/Secure.c:532:9:[m[K [01;35m[Kwarning: [m[K'[01m[KEVP_PKEY_get0_RSA[m[K' is deprecated: Since OpenSSL 3.0 [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wdeprecated-declarations-Wdeprecated-declarations]8;;[m[K]
  532 |         [01;35m[Krsa[m[K = (RSA *)EVP_PKEY_get0_RSA(k->pkey);
      |         [01;35m[K^~~[m[K
In file included from [01m[K/nix/store/pp2zf8bdgyz60ds8vcshk2603gcjgp72-openssl-3.3.2-dev/include/openssl/x509.h:29[m[K,
                 from [01m[K/nix/store/pp2zf8bdgyz60ds8vcshk2603gcjgp72-openssl-3.3.2-dev/include/openssl/ssl.h:32[m[K,
                 from [01m[Ksrc/Mayaqua/Secure.c:121[m[K:
[01m[K/nix/store/pp2zf8bdgyz60ds8vcshk2603gcjgp72-openssl-3.3.2-dev/include/openssl/evp.h:1356:22:[m[K [01;36m[Knote: [m[Kdeclared here
 1356 | const struct rsa_st *[01;36m[KEVP_PKEY_get0_RSA[m[K(const EVP_PKEY *pkey);
      |                      [01;36m[K^~~~~~~~~~~~~~~~~[m[K
[01m[Ksrc/Mayaqua/Secure.c:540:9:[m[K [01;35m[Kwarning: [m[K'[01m[KRSA_get0_key[m[K' is deprecated: Since OpenSSL 3.0 [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wdeprecated-declarations-Wdeprecated-declarations]8;;[m[K]
  540 |         [01;35m[KRSA_get0_key[m[K(rsa, &n, &e, &d);
      |         [01;35m[K^~~~~~~~~~~~[m[K
In file included from [01m[K/nix/store/pp2zf8bdgyz60ds8vcshk2603gcjgp72-openssl-3.3.2-dev/include/openssl/x509.h:36[m[K:
[01m[K/nix/store/pp2zf8bdgyz60ds8vcshk2603gcjgp72-openssl-3.3.2-dev/include/openssl/rsa.h:228:28:[m[K [01;36m[Knote: [m[Kdeclared here
  228 | OSSL_DEPRECATEDIN_3_0 void [01;36m[KRSA_get0_key[m[K(const RSA *r,
      |                            [01;36m[K^~~~~~~~~~~~[m[K
[01m[Ksrc/Mayaqua/Secure.c:541:9:[m[K [01;35m[Kwarning: [m[K'[01m[KRSA_get0_factors[m[K' is deprecated: Since OpenSSL 3.0 [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wdeprecated-declarations-Wdeprecated-declarations]8;;[m[K]
  541 |         [01;35m[KRSA_get0_factors[m[K(rsa, &p, &q);
      |         [01;35m[K^~~~~~~~~~~~~~~~[m[K
[01m[K/nix/store/pp2zf8bdgyz60ds8vcshk2603gcjgp72-openssl-3.3.2-dev/include/openssl/rsa.h:231:28:[m[K [01;36m[Knote: [m[Kdeclared here
  231 | OSSL_DEPRECATEDIN_3_0 void [01;36m[KRSA_get0_factors[m[K(const RSA *r,
      |                            [01;36m[K^~~~~~~~~~~~~~~~[m[K
[01m[Ksrc/Mayaqua/Secure.c:542:9:[m[K [01;35m[Kwarning: [m[K'[01m[KRSA_get0_crt_params[m[K' is deprecated: Since OpenSSL 3.0 [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wdeprecated-declarations-Wdeprecated-declarations]8;;[m[K]
  542 |         [01;35m[KRSA_get0_crt_params[m[K(rsa, &dmp1, &dmq1, &iqmp);
      |         [01;35m[K^~~~~~~~~~~~~~~~~~~[m[K
[01m[K/nix/store/pp2zf8bdgyz60ds8vcshk2603gcjgp72-openssl-3.3.2-dev/include/openssl/rsa.h:236:28:[m[K [01;36m[Knote: [m[Kdeclared here
  236 | OSSL_DEPRECATEDIN_3_0 void [01;36m[KRSA_get0_crt_params[m[K(const RSA *r,
      |                            [01;36m[K^~~~~~~~~~~~~~~~~~~[m[K
[01m[Ksrc/Mayaqua/Secure.c:601:90:[m[K [01;31m[Kerror: [m[Kpassing argument 4 of '[01m[Ksec->Api->C_CreateObject[m[K' from incompatible pointer type [[01;31m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wincompatible-pointer-types-Wincompatible-pointer-types]8;;[m[K]
  601 |         if ((ret = sec->Api->C_CreateObject(sec->SessionId, a, sizeof(a) / sizeof(a[0]), [01;31m[K&object[m[K)) != CKR_OK)
      |                                                                                          [01;31m[K^~~~~~~[m[K
      |                                                                                          [01;31m[K|[m[K
      |                                                                                          [01;31m[KUINT * {aka unsigned int *}[m[K
[01m[Ksrc/Mayaqua/Secure.c:601:90:[m[K [01;36m[Knote: [m[Kexpected '[01m[KCK_OBJECT_HANDLE_PTR[m[K' {aka '[01m[Klong unsigned int *[m[K'} but argument is of type '[01m[KUINT *[m[K' {aka '[01m[Kunsigned int *[m[K'}
[01m[Ksrc/Mayaqua/Secure.c:[m[K In function '[01m[KWriteSecCert[m[K':
[01m[Ksrc/Mayaqua/Secure.c:807:90:[m[K [01;31m[Kerror: [m[Kpassing argument 4 of '[01m[Ksec->Api->C_CreateObject[m[K' from incompatible pointer type [[01;31m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wincompatible-pointer-types-Wincompatible-pointer-types]8;;[m[K]
  807 |         if ((ret = sec->Api->C_CreateObject(sec->SessionId, a, sizeof(a) / sizeof(a[0]), [01;31m[K&object[m[K)) != CKR_OK)
      |                                                                                          [01;31m[K^~~~~~~[m[K
      |                                                                                          [01;31m[K|[m[K
      |                                                                                          [01;31m[KUINT * {aka unsigned int *}[m[K
[01m[Ksrc/Mayaqua/Secure.c:807:90:[m[K [01;36m[Knote: [m[Kexpected '[01m[KCK_OBJECT_HANDLE_PTR[m[K' {aka '[01m[Klong unsigned int *[m[K'} but argument is of type '[01m[KUINT *[m[K' {aka '[01m[Kunsigned int *[m[K'}
[01m[Ksrc/Mayaqua/Secure.c:[m[K In function '[01m[KEnumSecObject[m[K':
[01m[Ksrc/Mayaqua/Secure.c:1313:53:[m[K [01;31m[Kerror: [m[Kpassing argument 2 of '[01m[Ksec->Api->C_FindObjects[m[K' from incompatible pointer type [[01;31m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wincompatible-pointer-types-Wincompatible-pointer-types]8;;[m[K]
 1313 |         if (sec->Api->C_FindObjects(sec->SessionId, [01;31m[Kobjects[m[K, sizeof(objects) / sizeof(objects[0]), &num_objects) != CKR_OK)
      |                                                     [01;31m[K^~~~~~~[m[K
      |                                                     [01;31m[K|[m[K
      |                                                     [01;31m[KUINT * {aka unsigned int *}[m[K
[01m[Ksrc/Mayaqua/Secure.c:1313:53:[m[K [01;36m[Knote: [m[Kexpected '[01m[KCK_OBJECT_HANDLE_PTR[m[K' {aka '[01m[Klong unsigned int *[m[K'} but argument is of type '[01m[KUINT *[m[K' {aka '[01m[Kunsigned int *[m[K'}
[01m[Ksrc/Mayaqua/Secure.c:1313:100:[m[K [01;31m[Kerror: [m[Kpassing argument 4 of '[01m[Ksec->Api->C_FindObjects[m[K' from incompatible pointer type [[01;31m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wincompatible-pointer-types-Wincompatible-pointer-types]8;;[m[K]
 1313 |         if (sec->Api->C_FindObjects(sec->SessionId, objects, sizeof(objects) / sizeof(objects[0]), [01;31m[K&num_objects[m[K) != CKR_OK)
      |                                                                                                    [01;31m[K^~~~~~~~~~~~[m[K
      |                                                                                                    [01;31m[K|[m[K
      |                                                                                                    [01;31m[KUINT * {aka unsigned int *}[m[K
[01m[Ksrc/Mayaqua/Secure.c:1313:100:[m[K [01;36m[Knote: [m[Kexpected '[01m[KCK_ULONG_PTR[m[K' {aka '[01m[Klong unsigned int *[m[K'} but argument is of type '[01m[KUINT *[m[K' {aka '[01m[Kunsigned int *[m[K'}
[01m[Ksrc/Mayaqua/Secure.c:[m[K In function '[01m[KWriteSecData[m[K':
[01m[Ksrc/Mayaqua/Secure.c:1434:83:[m[K [01;31m[Kerror: [m[Kpassing argument 4 of '[01m[Ksec->Api->C_CreateObject[m[K' from incompatible pointer type [[01;31m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wincompatible-pointer-types-Wincompatible-pointer-types]8;;[m[K]
 1434 |         if (sec->Api->C_CreateObject(sec->SessionId, a, sizeof(a) / sizeof(a[0]), [01;31m[K&object[m[K) != CKR_OK)
      |                                                                                   [01;31m[K^~~~~~~[m[K
      |                                                                                   [01;31m[K|[m[K
      |                                                                                   [01;31m[KUINT * {aka unsigned int *}[m[K
[01m[Ksrc/Mayaqua/Secure.c:1434:83:[m[K [01;36m[Knote: [m[Kexpected '[01m[KCK_OBJECT_HANDLE_PTR[m[K' {aka '[01m[Klong unsigned int *[m[K'} but argument is of type '[01m[KUINT *[m[K' {aka '[01m[Kunsigned int *[m[K'}
[01m[Ksrc/Mayaqua/Secure.c:[m[K In function '[01m[KOpenSecSession[m[K':
[01m[Ksrc/Mayaqua/Secure.c:1737:66:[m[K [01;31m[Kerror: [m[Kpassing argument 5 of '[01m[Ksec->Api->C_OpenSession[m[K' from incompatible pointer type [[01;31m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wincompatible-pointer-types-Wincompatible-pointer-types]8;;[m[K]
 1737 |                 CKF_RW_SESSION | CKF_SERIAL_SESSION, NULL, NULL, [01;31m[K&session[m[K)) != CKR_OK)
      |                                                                  [01;31m[K^~~~~~~~[m[K
      |                                                                  [01;31m[K|[m[K
      |                                                                  [01;31m[KUINT * {aka unsigned int *}[m[K
[01m[Ksrc/Mayaqua/Secure.c:1737:66:[m[K [01;36m[Knote: [m[Kexpected '[01m[KCK_SESSION_HANDLE_PTR[m[K' {aka '[01m[Klong unsigned int *[m[K'} but argument is of type '[01m[KUINT *[m[K' {aka '[01m[Kunsigned int *[m[K'}
[01m[Ksrc/Mayaqua/Secure.c:1742:57:[m[K [01;31m[Kerror: [m[Kpassing argument 5 of '[01m[Ksec->Api->C_OpenSession[m[K' from incompatible pointer type [[01;31m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wincompatible-pointer-types-Wincompatible-pointer-types]8;;[m[K]
 1742 |                         CKF_SERIAL_SESSION, NULL, NULL, [01;31m[K&session[m[K)) != CKR_OK)
      |                                                         [01;31m[K^~~~~~~~[m[K
      |                                                         [01;31m[K|[m[K
      |                                                         [01;31m[KUINT * {aka unsigned int *}[m[K
[01m[Ksrc/Mayaqua/Secure.c:1742:57:[m[K [01;36m[Knote: [m[Kexpected '[01m[KCK_SESSION_HANDLE_PTR[m[K' {aka '[01m[Klong unsigned int *[m[K'} but argument is of type '[01m[KUINT *[m[K' {aka '[01m[Kunsigned int *[m[K'}
[01m[Ksrc/Mayaqua/Secure.c:[m[K In function '[01m[KOpenSec[m[K':
[01m[Ksrc/Mayaqua/Secure.c:1830:56:[m[K [01;31m[Kerror: [m[Kpassing argument 3 of '[01m[Ksec->Api->C_GetSlotList[m[K' from incompatible pointer type [[01;31m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wincompatible-pointer-types-Wincompatible-pointer-types]8;;[m[K]
 1830 |         if ((err = sec->Api->C_GetSlotList(true, NULL, [01;31m[K&sec->NumSlot[m[K)) != CKR_OK || sec->NumSlot == 0)
      |                                                        [01;31m[K^~~~~~~~~~~~~[m[K
      |                                                        [01;31m[K|[m[K
      |                                                        [01;31m[KUINT * {aka unsigned int *}[m[K
[01m[Ksrc/Mayaqua/Secure.c:1830:56:[m[K [01;36m[Knote: [m[Kexpected '[01m[KCK_ULONG_PTR[m[K' {aka '[01m[Klong unsigned int *[m[K'} but argument is of type '[01m[KUINT *[m[K' {aka '[01m[Kunsigned int *[m[K'}
[01m[Ksrc/Mayaqua/Secure.c:1840:46:[m[K [01;31m[Kerror: [m[Kpassing argument 2 of '[01m[Ksec->Api->C_GetSlotList[m[K' from incompatible pointer type [[01;31m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wincompatible-pointer-types-Wincompatible-pointer-types]8;;[m[K]
 1840 |         if (sec->Api->C_GetSlotList(TRUE, [01;31m[Ksec->SlotIdList[m[K, &sec->NumSlot) != CKR_OK)
      |                                           [01;31m[K~~~^~~~~~~~~~~~[m[K
      |                                              [01;31m[K|[m[K
      |                                              [01;31m[KUINT * {aka unsigned int *}[m[K
[01m[Ksrc/Mayaqua/Secure.c:1840:46:[m[K [01;36m[Knote: [m[Kexpected '[01m[KCK_SLOT_ID_PTR[m[K' {aka '[01m[Klong unsigned int *[m[K'} but argument is of type '[01m[KUINT *[m[K' {aka '[01m[Kunsigned int *[m[K'}
[01m[Ksrc/Mayaqua/Secure.c:1840:60:[m[K [01;31m[Kerror: [m[Kpassing argument 3 of '[01m[Ksec->Api->C_GetSlotList[m[K' from incompatible pointer type [[01;31m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wincompatible-pointer-types-Wincompatible-pointer-types]8;;[m[K]
 1840 |         if (sec->Api->C_GetSlotList(TRUE, sec->SlotIdList, [01;31m[K&sec->NumSlot[m[K) != CKR_OK)
      |                                                            [01;31m[K^~~~~~~~~~~~~[m[K
      |                                                            [01;31m[K|[m[K
      |                                                            [01;31m[KUINT * {aka unsigned int *}[m[K
[01m[Ksrc/Mayaqua/Secure.c:1840:60:[m[K [01;36m[Knote: [m[Kexpected '[01m[KCK_ULONG_PTR[m[K' {aka '[01m[Klong unsigned int *[m[K'} but argument is of type '[01m[KUINT *[m[K' {aka '[01m[Kunsigned int *[m[K'}
make: *** [Makefile:147: tmp/objs/Mayaqua/Secure.o] Error 1
```

### Additional context

It seems to fail due to a warning being treated as error:
```
src/Mayaqua/Secure.c:1840:60: error: passing argument 3 of 'sec->Api->C_GetSlotList' from incompatible pointer type [-Wincompatible-pointer-types]
 1840 |         if (sec->Api->C_GetSlotList(TRUE, sec->SlotIdList, &sec->NumSlot) != CKR_OK)
      |                                                            ^~~~~~~~~~~~~
      |                                                            |
      |                                                            UINT * {aka unsigned int *}
src/Mayaqua/Secure.c:1840:60: note: expected 'CK_ULONG_PTR' {aka 'long unsigned int *'} but argument is of type 'UINT *' {aka 'unsigned int *'}
```

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.71, NixOS, 25.05 (Warbler), 25.05.20250116.5df4362`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Lix, like Nix) 2.91.1
System type: x86_64-linux
Additional system types: i686-linux, x86_64-v1-linux, x86_64-v2-linux, x86_64-v3-linux, x86_64-v4-linux
Features: gc, signed-caches
System configuration file: /etc/nix/nix.conf
User configuration files: /home/mat/.config/nix/nix.conf:/nix/store/8m5j7mvpvjwwxsaklmpa0y8291i4ki6g-stylix-kde-config/nix/nix.conf:/etc/xdg/nix/nix.conf:/home/mat/.local/share/flatpak/exports/etc/xdg/nix/nix.conf:/var/lib/flatpak/exports/etc/xdg/nix/nix.conf:/home/mat/.nix-profile/etc/xdg/nix/nix.conf:/nix/profile/etc/xdg/nix/nix.conf:/home/mat/.local/state/nix/profile/etc/xdg/nix/nix.conf:/etc/profiles/per-user/mat/etc/xdg/nix/nix.conf:/nix/var/nix/profiles/default/etc/xdg/nix/nix.conf:/run/current-system/sw/etc/xdg/nix/nix.conf
Store directory: /nix/store
State directory: /nix/var/nix
Data directory: /nix/store/6dxshp2qaas92v7d36baf7y95q0v98dn-lix-2.91.1/share`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/store/l9nb64iii15y0nr37qrs1cfm6rlpg6gh-source`



### Notify maintainers

@rick68 
---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/so/softether/package.nix
+++ b/pkgs/by-name/so/softether/package.nix
@@ -6,6 +6,7 @@
   readline,
   ncurses,
   zlib,
+  bash,
   dataDir ? "/var/lib/softether",
 }:
 
@@ -25,6 +26,7 @@ stdenv.mkDerivation (finalAttrs: {
     readline
     ncurses
     zlib
+    bash
   ];
 
   preConfigure = ''

--- a/pkgs/by-name/so/softether/package.nix
+++ b/pkgs/by-name/so/softether/package.nix
@@ -41,6 +41,13 @@ stdenv.mkDerivation (finalAttrs: {
       Makefile
   '';
 
+  postInstall = ''
+    substituteInPlace $out/bin/vpnbridge --replace-fail /var/lib/softether/vpnbridge/vpnbridge $out/var/lib/softether/vpnbridge/vpnbridge 
+    substituteInPlace $out/bin/vpnclient --replace-fail /var/lib/softether/vpnclient/vpnclient $out/var/lib/softether/vpnclient/vpnclient 
+    substituteInPlace $out/bin/vpncmd --replace-fail /var/lib/softether/vpncmd/vpncmd $out/var/lib/softether/vpncmd/vpncmd 
+    substituteInPlace $out/bin/vpnserver --replace-fail /var/lib/softether/vpnserver/vpnserver $out/var/lib/softether/vpnserver/vpnserver 
+  '';
+
   env.NIX_CFLAGS_COMPILE = toString [
     "-Wno-incompatible-pointer-types"
     "-Wno-implicit-function-declaration"

--- a/pkgs/by-name/so/softether/package.nix
+++ b/pkgs/by-name/so/softether/package.nix
@@ -41,6 +41,11 @@ stdenv.mkDerivation (finalAttrs: {
       Makefile
   '';
 
+  env.NIX_CFLAGS_COMPILE = toString [
+    "-Wno-incompatible-pointer-types"
+    "-Wno-implicit-function-declaration"
+  ];
+
   meta = {
     description = "Open-Source Free Cross-platform Multi-protocol VPN Program";
     homepage = "https://www.softether.org/";

--- a/pkgs/by-name/so/softether/package.nix
+++ b/pkgs/by-name/so/softether/package.nix
@@ -44,10 +44,10 @@ stdenv.mkDerivation (finalAttrs: {
   '';
 
   postInstall = ''
-    substituteInPlace $out/bin/vpnbridge --replace-fail /var/lib/softether/vpnbridge/vpnbridge $out/var/lib/softether/vpnbridge/vpnbridge 
-    substituteInPlace $out/bin/vpnclient --replace-fail /var/lib/softether/vpnclient/vpnclient $out/var/lib/softether/vpnclient/vpnclient 
-    substituteInPlace $out/bin/vpncmd --replace-fail /var/lib/softether/vpncmd/vpncmd $out/var/lib/softether/vpncmd/vpncmd 
-    substituteInPlace $out/bin/vpnserver --replace-fail /var/lib/softether/vpnserver/vpnserver $out/var/lib/softether/vpnserver/vpnserver 
+    substituteInPlace $out/bin/vpnbridge --replace-fail /var/lib/softether/vpnbridge/vpnbridge $out/var/lib/softether/vpnbridge/vpnbridge
+    substituteInPlace $out/bin/vpnclient --replace-fail /var/lib/softether/vpnclient/vpnclient $out/var/lib/softether/vpnclient/vpnclient
+    substituteInPlace $out/bin/vpncmd --replace-fail /var/lib/softether/vpncmd/vpncmd $out/var/lib/softether/vpncmd/vpncmd
+    substituteInPlace $out/bin/vpnserver --replace-fail /var/lib/softether/vpnserver/vpnserver $out/var/lib/softether/vpnserver/vpnserver
   '';
 
   env.NIX_CFLAGS_COMPILE = toString [

